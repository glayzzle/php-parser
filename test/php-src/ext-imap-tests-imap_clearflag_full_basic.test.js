// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/imap/tests/imap_clearflag_full_basic.phpt
  it("Test imap_clearflag_full() function : basic functionality", function () {
    expect(parser.parseCode("<?php\necho \"*** Testing imap_clearflag_full() : basic functionality ***\\n\";\nrequire_once(__DIR__.'/setup/imap_include.inc');\necho \"Create a new mailbox for test\\n\";\n$imap_stream = setup_test_mailbox(\"imapclearflagfullbasic\", 10);\n$check = imap_check($imap_stream);\necho \"Initial msg count in new_mailbox : \". $check->Nmsgs . \"\\n\";\necho \"Set some flags\\n\";\nvar_dump(imap_setflag_full($imap_stream, \"1,3\", \"\\\\Seen \\\\Answered\"));\nvar_dump(imap_setflag_full($imap_stream, \"2,4\", \"\\\\Answered\"));\nvar_dump(imap_setflag_full($imap_stream, \"5,7\", \"\\\\Flagged \\\\Deleted\"));\nvar_dump(imap_setflag_full($imap_stream, \"6,8\", \"\\\\Deleted\"));\nvar_dump(imap_setflag_full($imap_stream, \"9,10\", \"\\\\Draft \\\\Flagged\"));\nvar_dump(imap_search($imap_stream, \"SEEN\"));\nvar_dump(imap_search($imap_stream, \"ANSWERED\"));\nvar_dump(imap_search($imap_stream, \"FLAGGED\"));\nvar_dump(imap_search($imap_stream, \"DELETED\"));\nvar_dump(imap_clearflag_full($imap_stream, \"1,4\", \"\\\\Answered\"));\nvar_dump(imap_clearflag_full($imap_stream, \"5,6,7,8\", \"\\\\Deleted\"));\nvar_dump(imap_clearflag_full($imap_stream, \"9\", \"\\\\Flagged\"));\nvar_dump(imap_search($imap_stream, \"SEEN\"));\nvar_dump(imap_search($imap_stream, \"ANSWERED\"));\nvar_dump(imap_search($imap_stream, \"FLAGGED\"));\nvar_dump(imap_search($imap_stream, \"DELETED\"));\nimap_close($imap_stream);\n?>")).toMatchSnapshot();
  });
});
