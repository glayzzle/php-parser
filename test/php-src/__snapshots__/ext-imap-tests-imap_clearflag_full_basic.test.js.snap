// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Test imap_clearflag_full() function : basic functionality 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"*** Testing imap_clearflag_full() : basic functionality ***\\\\n\\"",
          "unicode": false,
          "value": "*** Testing imap_clearflag_full() : basic functionality ***
",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
    ExpressionStatement {
      "expression": Include {
        "kind": "include",
        "once": true,
        "require": true,
        "target": Bin {
          "kind": "bin",
          "left": Magic {
            "kind": "magic",
            "raw": "__DIR__",
            "value": "__DIR__",
          },
          "parenthesizedExpression": true,
          "right": String {
            "isDoubleQuote": false,
            "kind": "string",
            "raw": "'/setup/imap_include.inc'",
            "unicode": false,
            "value": "/setup/imap_include.inc",
          },
          "type": ".",
        },
      },
      "kind": "expressionstatement",
    },
    Echo {
      "expressions": Array [
        String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"Create a new mailbox for test\\\\n\\"",
          "unicode": false,
          "value": "Create a new mailbox for test
",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "imap_stream",
        },
        "operator": "=",
        "right": Call {
          "arguments": Array [
            String {
              "isDoubleQuote": true,
              "kind": "string",
              "raw": "\\"imapclearflagfullbasic\\"",
              "unicode": false,
              "value": "imapclearflagfullbasic",
            },
            Number {
              "kind": "number",
              "value": "10",
            },
          ],
          "kind": "call",
          "what": Name {
            "kind": "name",
            "name": "setup_test_mailbox",
            "resolution": "uqn",
          },
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "check",
        },
        "operator": "=",
        "right": Call {
          "arguments": Array [
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "imap_stream",
            },
          ],
          "kind": "call",
          "what": Name {
            "kind": "name",
            "name": "imap_check",
            "resolution": "uqn",
          },
        },
      },
      "kind": "expressionstatement",
    },
    Echo {
      "expressions": Array [
        Bin {
          "kind": "bin",
          "left": Bin {
            "kind": "bin",
            "left": String {
              "isDoubleQuote": true,
              "kind": "string",
              "raw": "\\"Initial msg count in new_mailbox : \\"",
              "unicode": false,
              "value": "Initial msg count in new_mailbox : ",
            },
            "right": PropertyLookup {
              "kind": "propertylookup",
              "offset": Identifier {
                "kind": "identifier",
                "name": "Nmsgs",
              },
              "what": Variable {
                "curly": false,
                "kind": "variable",
                "name": "check",
              },
            },
            "type": ".",
          },
          "right": String {
            "isDoubleQuote": true,
            "kind": "string",
            "raw": "\\"\\\\n\\"",
            "unicode": false,
            "value": "
",
          },
          "type": ".",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
    Echo {
      "expressions": Array [
        String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"Set some flags\\\\n\\"",
          "unicode": false,
          "value": "Set some flags
",
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"1,3\\"",
                "unicode": false,
                "value": "1,3",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Seen \\\\\\\\Answered\\"",
                "unicode": false,
                "value": "\\\\Seen \\\\Answered",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_setflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"2,4\\"",
                "unicode": false,
                "value": "2,4",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Answered\\"",
                "unicode": false,
                "value": "\\\\Answered",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_setflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"5,7\\"",
                "unicode": false,
                "value": "5,7",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Flagged \\\\\\\\Deleted\\"",
                "unicode": false,
                "value": "\\\\Flagged \\\\Deleted",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_setflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"6,8\\"",
                "unicode": false,
                "value": "6,8",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Deleted\\"",
                "unicode": false,
                "value": "\\\\Deleted",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_setflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"9,10\\"",
                "unicode": false,
                "value": "9,10",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Draft \\\\\\\\Flagged\\"",
                "unicode": false,
                "value": "\\\\Draft \\\\Flagged",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_setflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"SEEN\\"",
                "unicode": false,
                "value": "SEEN",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"ANSWERED\\"",
                "unicode": false,
                "value": "ANSWERED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"FLAGGED\\"",
                "unicode": false,
                "value": "FLAGGED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"DELETED\\"",
                "unicode": false,
                "value": "DELETED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"1,4\\"",
                "unicode": false,
                "value": "1,4",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Answered\\"",
                "unicode": false,
                "value": "\\\\Answered",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_clearflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"5,6,7,8\\"",
                "unicode": false,
                "value": "5,6,7,8",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Deleted\\"",
                "unicode": false,
                "value": "\\\\Deleted",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_clearflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"9\\"",
                "unicode": false,
                "value": "9",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"\\\\\\\\Flagged\\"",
                "unicode": false,
                "value": "\\\\Flagged",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_clearflag_full",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"SEEN\\"",
                "unicode": false,
                "value": "SEEN",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"ANSWERED\\"",
                "unicode": false,
                "value": "ANSWERED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"FLAGGED\\"",
                "unicode": false,
                "value": "FLAGGED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Variable {
                "curly": false,
                "kind": "variable",
                "name": "imap_stream",
              },
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"DELETED\\"",
                "unicode": false,
                "value": "DELETED",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "imap_search",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Variable {
            "curly": false,
            "kind": "variable",
            "name": "imap_stream",
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "imap_close",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
